File: src/main/java/com/example/wizytydomowe/Appointment/Appointment.java
package com.example.wizytydomowe.Appointment;

import com.example.wizytydomowe.Doctor.Doctor;
import com.example.wizytydomowe.Patient.Patient;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Digits;
import javax.validation.constraints.Future;
import java.time.LocalDate;

@Entity
@Data
@Table(name = "appointment")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Future
    private LocalDate date;
    private Importance importance;
    @Digits(integer = 4, fraction = 2)
    private Integer price;
    private String comments;
    private String specialization;
    @ManyToOne
    @JoinColumn(name = "doctor_id")
    private Doctor doctor;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "patient_id")
    private Patient patient;

}

File: client/src/app/appointment.ts
import {Patient} from "./patient";

export class Appointment {
  importance: string = '';
  specialization: string = '';
  comments: string = '';
  patient: Patient;

  constructor(importance: string, specialization: string,
              comments: string, patient: Patient) {
    this.importance = importance;
    this.specialization = specialization;
    this.comments = comments;
    this.patient = patient;
  }
}
